File: smartmoneyserver/src/main/java/fpt/aptech/server/enums/debt/DebtType.java
package fpt.aptech.server.enums.debt;

import lombok.Getter;

/**
 * Định nghĩa loại nợ trong sổ nợ.
 * Dùng cho tDebts.debt_type
 */
@Getter
public enum DebtType {
    PAYABLE(0),         // Khoản phải trả (Đi vay)
    RECEIVABLE(1);      // Khoản phải thu (Cho vay)

    private final int value;

    DebtType(int value) {
        this.value = value;
    }
}

File: APP MOBILE/smart_money/lib/modules/debt/enums/debt_type.dart
/// Loại nợ: Phải trả (Đi vay) hoặc Phải thu (Cho vay)
enum DebtType {
  payable,   // 0 - Phải trả (Đi vay)
  receivable; // 1 - Phải thu (Cho vay)

  int get value => index;

  static DebtType fromValue(int value) {
    return DebtType.values[value];
  }

  static DebtType? fromValueOrNull(int? value) {
    if (value == null || value < 0 || value >= DebtType.values.length) {
      return null;
    }
    return DebtType.values[value];
  }

  String get displayName {
    switch (this) {
      case DebtType.payable:
        return 'Cần trả (Đi vay)';
      case DebtType.receivable:
        return 'Cần thu (Cho vay)';
    }
  }
}

